### DISCUSSION TITLE
Deployment on GCP, contribute to deploy on GCP

### DISCUSSION THREAD

**Discussion by ai-from-70s** [matanolabs/matano#173] [Ideas]
We are running on GCP stack which have I believe equivalent set technologies used by Matano:

S3 = GCS 
Lambda =  GCP Cloud Functions
SNS/SQS = GCP Cloud Pub/Sub

 I am really interested in adding ability to migrate/run your system on GCP (either as a fork or as contribution to the main repo)

What do you think is the effort to write adapters for the GCP components, is it a major rewrite or is architecture modular where the interfaces exists for each of the components and it's a matter of adding cloud specific implementations and tests?

**Comment by Samrose-Ahmed** [comment]
Yes and we can use BigQuery got analytics with BigLake.

This is planned although a medium term goal, it works require effort in rewriting all the infrastructure and hooking it up with the code.
